### ISSUE TITLE
import matplotlib.pyplot fails on ubuntu

### ISSUE THREAD

**Issue by kbroughton** [ContinuumIO/anaconda-issues#244]
This worked fine on a fresh install to mac yosemite.
On ubuntu, I get an import error trying to use matplotlib.pyplot

*****SOLUTION
*sudo apt-get install python-qt4
*****

step to reproduce

1. install anaconda
2. ipython notebook
3. import matplotlib.pyplot
output 

/home/ubuntu/anaconda/lib/python2.7/site-packages/matplotlib/backends/qt_compat.py in <module>()
     74     if QT_API in [QT_API_PYQT, QT_API_PYQTv2]:  # PyQt4 API
     75 
---> 76         from PyQt4 import QtCore, QtGui
     77 
     78         try:

ImportError: libSM.so.6: cannot open shared object file: No such file or directory

I searched using locate libSm and found nothing on the ubuntu  machine.
So i installed this with 
sudo apt-get install libsm6
and I symlinked it in ~/anaconda/lib for good measure.

This fixed the first error.  Continuing on I got
/home/ubuntu/anaconda/lib/python2.7/site-packages/matplotlib/backends/qt_compat.py in <module>()
     74     if QT_API in [QT_API_PYQT, QT_API_PYQTv2]:  # PyQt4 API
     75 
---> 76         from PyQt4 import QtCore, QtGui
     77 
     78         try:

ImportError: libXrender.so.1: cannot open shared object file: No such file or directory

At which point i guessed the whole pyqt4 might be missing

SOLUTION

sudo apt-get install python-qt4

Now import matplotlib.pyplot works

**Comment by ccordoba12** [comment]
Strange error report. Closing because the OP reports he fixed it.

**Comment by kevleininger** [comment]
You shouldn't close this.  You should fix it.  I got the same error.  Shouldn't this be part of the anaconda package?

**Comment by msarahan** [comment]
Sorry, but the solution posted here points strongly to the user environment being incorrect somehow.  PyQt4 and Qt4 are both part of Anaconda, and we do test anaconda releases on Ubuntu, among many other platforms.

Do you have LD_LIBRARY_PATH, PYTHONPATH, or any other critical PATH-like environment variables set?  These could lead to behavior described here.

**Comment by Rumpkie** [comment]
Sorry to disagree @msarahan, ran into the same problem on a clean Ubuntu 16.04 server install today, and the solution is still valid.

**Comment by msarahan** [comment]
Ubuntu server generally does not include a GUI, does it?  Anaconda sort of assumes (but unfortunately, does not document) that you have a fundamental GUI environment set up.  Please tell us more about your Ubuntu setup.

**Comment by Rumpkie** [comment]
@msarahan you are correct, Ubuntu server does not have a GUI install by default and I'm sure that's where the issue is. While Anaconda did have its qt libraries installed, the system did not.

I have installed Anaconda on several Ubuntu 16.04 desktops and haven't had the issue.  

I'm playing around with matplotlib on jupyter notebooks. The Ubuntu system is being hosted GUI-less on a micro cloud.

This is a vanilla install of Ubuntu Server 16.04, Open SSH, python-qt4, and anaconda with jupyter, matplotlib, and numpy.

I had to also set:
c.IPKernelApp.matplotlib = u'inline'  in the jupyter config (I don't think this is working) and
add
import matplotlib
matplotlib.use('Agg')
%matplotlib inline (the config line was working right, I should be able to get by without this)
to my notebooks if anybody wants the full setup.
